src/lib/rest/curl_context.cpp

```diff
@@ -0,0 +1,94 @@
+#include "curl_context.h"
+
+#include <map>
+
+#include <assert.h>
+
+#include "logMsg/logMsg.h"
+#include "logMsg/traceLevels.h"
+
+
+static pthread_mutex_t contexts_mutex = PTHREAD_MUTEX_INITIALIZER;
+static std::map<std::string, struct curl_context> contexts;
+
+struct curl_context
+get_curl_context(const std::string& key)
+{
+  struct curl_context cc = {NULL, NULL};
+  int s = pthread_mutex_lock(&contexts_mutex);
+  if(s!=0)
+  {
+       LM_X(1,("pthread_mutex_lock"));
```

Comment:
I just read now ...
1. I'll do the same you're doing in the other code, "when in Rome, do as the Romans do"
2. I disagree, "never ever" is too 'much'. I think it depends on the kind of error (recoverable or not, a sane action could be run or not, ...)
3. I think the current way some errors are being handled (or not handled to be precise) is completely WRONG and is FAR WORSE than dying when a mutex cannot be initialized or memory cannot be allocated, what I think it's completely sane (however 1. Romans rules)